components/base/kinematicbase/differentialDrive.go

```diff
@@ -320,3 +309,46 @@ func (ddk *differentialDriveKinematics) newValidRegionCapsule(starting, desired
 
 	return capsule, nil
 }
+
+//nolint: dupl
+func (ddk *differentialDriveKinematics) ErrorState(
+	ctx context.Context,
+	plan [][]referenceframe.Input,
+	currentNode int,
+) (spatialmath.Pose, error) {
+	if currentNode < 0 || currentNode >= len(plan) {
+		return nil, fmt.Errorf("cannot get ErrorState for node %d, must be >= 0 and less than plan length %d", currentNode, len(plan))
+	}
+
+	// Get pose-in-frame of the base via its localizer. The offset between the localizer and its base should already be accounted for.
+	actualPIF, err := ddk.CurrentPosition(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var nominalPose spatialmath.Pose
+
+	// Determine the nominal pose, that is, the pose where the robot ought be if it had followed the plan perfectly up until this point.
+	// This is done differently depending on what sort of frame we are working with.
+	if len(plan) < 2 {
+		return nil, errors.New("diff drive motion plan must have at least two waypoints")
+	}
+	nominalPose, err = ddk.planningFrame.Transform(plan[currentNode])
+	if err != nil {
+		return nil, err
+	}
+	if currentNode > 0 {
```

Comment:
I'm not sure this ever is zero if we don't have plans that we call the first waypoint as the first element rather than the zeroth.  Could modify the error check at the top of the function to account for this and then remove this conditional.  

I'm not sure it works correctly as written if current node is zero.

Reply:
You're right; since we need to look at the previous point to really determine nominal position, this can't be used on the 0th point. Added an error.